cadc-storage-adapter-fs/src/main/java/org/opencadc/inventory/storage/fs/FileSystemStorageAdapter.java

```diff
@@ -582,5 +590,18 @@ private Path createStorageLocationPath(StorageLocation storageLocation) {
         Path ret = root.resolve(path.toString());
         return ret;
     }
-    
+
+    public static void setFileAttribute(Path path, String propertyName, String property) throws IOException {
```

Comment:
would prefer args that was key-value or name-value pair... "property" is often sometimes used to indicate the key but usually means the pair (key and value)

line ~599: calling getBytes(Charset) twice, but I don't see the value if logging the bytes when you already logged the string key = value

This method only needs one log.debug with all three method args - spreading it over two lines loses information.

And finally some ultra-picky optimisation: you can move then view creation inside the if statement

Reply:
done